src/classes/frDonor.cls

```diff
@@ -56,6 +56,7 @@ public class frDonor extends frModel implements frSyncable{
     }
     
     private Contact supporterContact;
+    private Account supporterAccount;
```

Comment:
Can you explain the need for this variable?  It does not appear to be populated currently which causes other issues in the place where the ID is being read

Reply:
Added the assignment for these 2 objects to allow the id to be read!